src/main/webapp/assets/js/signUpScript.js

```diff
@@ -52,23 +69,195 @@ async function onLoad() {
 
-  // When users select an option from the dropdown, set the college id.
-  document.getElementById('colleges-input-form').addEventListener('change', setCollegeID);
+  // When users submit the form, initiate form validation.
+  emailForm = document.getElementById('email-form');
+  submitFormButton = document.getElementById('form-submit');
+  subscribeFormSubmitted = document.getElementById('subscribe-form-submitted');
+  unsubscribeFormSubmitted = document.getElementById('unsubscribe-form-submitted');
+  submitFormButton.addEventListener('click', submitForm);
+}
+
+/**
+ * On click of the submit button, validate form data and send data to the servlet.
+ * @return {void}
+ */
+async function submitForm() {
+  // Disable multiple submissions.
+  submitFormButton.disabled = true;
+
+  if (validateForm()) {
+    emailForm.action = '/user';
+
+    // Based on user subscribing or unsubscribing, show confirmation that form
+    // was submitted.
+    if (document.getElementById('subscribe').checked) {
+      subscribeFormSubmitted.style.display = 'block';
+    } else {
+      unsubscribeFormSubmitted.style.display = 'block';
+    }
+
+    // Waits for 4000 ms so user can read confirmation and submits the form.
+    setTimeout(function() {
+      emailForm.submit();
+    }, 4000);
+  } else {
+    submitFormButton.disabled = false;
+  }
+}
+
+/**
+ * Goes through the form elements and marks the invalid inputs.
+ * Returns whether all the inputs are valid.
+ * @return {boolean}
+ */
+function validateForm() {
+  const invalidIds = [];
+  const errorMessages = [];
+  const formElements = emailForm.elements;
+
+  disableInjection(formElements);
+  validateFormText(invalidIds, errorMessages, formElements);
+  validateFormEmail(invalidIds, errorMessages, formElements);
+  validateFormCollege(invalidIds, errorMessages, formElements);
+  validateFormRadioButtons(invalidIds, errorMessages, formElements);
+  markInvalidInputs(invalidIds, errorMessages, formElements);
+
+  return (invalidIds.length === 0);
+}
+
+/**
+ * Checks if the email is valid.
+ * @param {array} invalidIds
+ * @param {array} errorMessages
+ * @param {array} formElements
+ * @return {void}
+ */
+function validateFormEmail(invalidIds, errorMessages, formElements) {
+  const email = formElements.namedItem('email').value;
+  const emailPattern = /^[a-zA-Z0-9._-]+@google.com$/;
+  if (!emailPattern.test(email)) {
+    invalidIds.push('email');
+    errorMessages.push('email must follow @google.com');
+  }
+}
+
+/**
+ * Checks if at least one radio button has been checked.
+ * @param {array} invalidIds
+ * @param {array} errorMessages
+ * @param {array} formElements
+ * @return {void}
+ */
+function validateFormRadioButtons(invalidIds, errorMessages, formElements) {
+  const subscribe = formElements.namedItem('email-notif')[0].checked;
+  const unsubscribe = formElements.namedItem('email-notif')[1].checked;
+
+  if (!subscribe && !unsubscribe ) {
```

Comment:
It's not clear what subscribe/unsubcribed mean here. Could you use something like `isSubscribed` or `shouldSubscribe`?

Reply:
This is Dart but should apply to other languages. https://dart.dev/guides/language/effective-dart/design#prefer-a-non-imperative-verb-phrase-for-a-boolean-property-or-variable

Reply:
It represents the radio button checked! Gotcha, will change!